TWTValidation/JSON Validator/Parser/TWTJSONSchemaParser.m

```diff
@@ -0,0 +1,652 @@
+}
+
+
+// Keywords: exclusiveMinimum & Maximum, uniqueItems
+- (BOOL)parseBooleanValueFromSchema:(NSDictionary *)schema forKey:(NSString *)key valueIfNotPresent:(BOOL)defaultValue
+{
+    [self pushPathComponent:key];
+    NSNumber *booleanObject = schema[key];
+    [self checkObject:booleanObject isOfClass:[NSNumber class] allowNil:YES];
+    [self popPathComponent];
+    return booleanObject ? [booleanObject boolValue] : defaultValue;
+}
+
+
+// Keywords: pattern
+- (NSString *)parseStringFromSchema:(NSDictionary *)schema forKey:(NSString *)key
+{
+    [self pushPathComponent:key];
+    NSString *string = schema[key];
+    [self checkObject:string isOfClass:[NSString class] allowNil:YES];
+    [self popPathComponent];
+    return string;
+}
+
+
+// Keywords: minimum, maximum
+- (NSNumber *)parseNumberFromSchema:(NSDictionary *)schema forKey:(NSString *)key
+{
+    [self pushPathComponent:key];
+    NSNumber *number = schema[key];
+    [self checkObject:number isOfClass:[NSNumber class] allowNil:YES];
+    [self popPathComponent];
+    return number;
+}
+
+
+// Keywords: multipleOf
+- (NSNumber *)parsePositiveNumberFromSchema:(NSDictionary *)schema forKey:(NSString *)key
+{
+    NSNumber *number = schema[key];
+    if (!number) {
+        return nil;
+    }
+
+    [self pushPathComponent:key];
+    [self checkObject:number isOfClass:[NSNumber class] allowNil:YES];
+    if ([number doubleValue] <= 0) {
+        NSString *reason = [NSString stringWithFormat:@"Invalid value for key %@. Expected number greater than zero.", key];
+        [self failWithErrorCode:TWTJSONSchemaParserErrorCodeInvalidValue reason:reason object:number];
+    }
+
+    [self popPathComponent];
+    return number;
+}
+
+
+// Keywords: min & maxLength, min & maxItems, min & maxProperties
+- (NSNumber *)parseUnsignedIntegerFromSchema:(NSDictionary *)schema forKey:(NSString *)key
+{
+    [self pushPathComponent:key];
+    NSNumber *number = schema[key];
+    [self checkObject:number isOfClass:[NSNumber class] allowNil:YES];
+
+    if (number) {
+        double numberValue = [number doubleValue];
+        BOOL isUnsignedInteger = numberValue >= 0 && numberValue == trunc(numberValue);
+        if (!isUnsignedInteger) {
+            NSString *reason = [NSString stringWithFormat:@"Invalid value for key %@. Expected unsigned integer.", key];
+            [self failWithErrorCode:TWTJSONSchemaParserErrorCodeInvalidValue reason:reason object:number];
+        }
+    }
+
+    [self popPathComponent];
+    return number;
+}
+
+
+// Keywords: definitions
+- (NSDictionary *)parseDictionaryFromSchema:(NSDictionary *)schema forKey:(NSString *)key
+{
+    [self pushPathComponent:key];
+    NSDictionary *dictionary = schema[key];
+    [self checkObject:dictionary isOfClass:[NSDictionary class] allowNil:YES];
+    [self popPathComponent];
+    return dictionary;
+}
+
+
+// Keywords: enum
+- (NSArray *)parseArrayFromSchema:(NSDictionary *)schema forKey:(NSString *)key
```

Comment:
Since this requires the array to be non-empty, I’d put that in the method name. `‑parseNonEmptyArrayForKey:schema:`.